File: Mongo-Nodejs-M101JS/Week 3/HW3.1/answer.js
var mongo = require('mongodb');

mongo.connect('mongodb://localhost:27017/school',function(err,db){
	if (err) throw err;

	_students = db.collection('students');

	_students.aggregate([
			{ '$unwind' : '$scores' },
			{'$match':{'scores.type':'homework'}},
			{"$group": {"_id": "$_id", "min": {"$min": '$scores.score'}}}
		],function(err,result){

			if (err) throw err;
			result.forEach(function(data){
				query = {'_id':data._id};
				_update = {'$scores.type':'homework','$pull':{'$score.score':data.min}};
				_students.update(query,_update,function(err,doc){
					console.dir('updated ' + doc );
				})
			});
			_students.aggregate([
						{'$unwind':'$scores'},
						{'$group':{'_id':'$_id','avg':{'$avg':'$scores.score'}}},
						{'$sort':{'avg':-1}},
						{'$limit':1}
					],function(err,result){
						console.dir("students has max avg score id :" + result.pop()._id);
						db.close();

			});
			}
	);

});

File: Mongo-Developer-M101P-Pyramid/Week 3/HW3.1/answer.py
import pymongo
from pymongo import MongoClient
from bson.son import SON
connection = MongoClient('localhost', 27017)

db = connection.school

students = db.students

item = students.aggregate([
	{ '$unwind' : '$scores' },
	{'$match':{'scores.type':'homework'}},
	{"$group": {"_id": "$_id", "min": {"$min": '$scores.score'}}}
	])
if item:
	for data in item['result']:
		query = {'_id':data['_id']}
		unset = {'$pull':{'scores':{'type':'homework','score':data['min']}}}
		students.update(query,unset)
print "completed remove data"
item = students.aggregate([
	{ '$unwind' : '$scores' },
	{"$group": {"_id": "$_id", "avg": {"$avg": '$scores.score'}}},
	{'$sort':SON([('avg',-1)])},
	])

print item['result'][0]
